switmt_to_iso20022/utils.bal

```diff
@@ -0,0 +1,1462 @@
+                {
+                    Inf: [rgltyRptg.substring(0)]
+                }
+            ]
+        }];
+    }
+    return ();
+}
+
+# Extracts and returns different parts of a party identifier based on its format.
+# The function handles different formats of the party identifier and returns an array with specific substrings
+# or `null` values depending on the conditions.
+#
+# + prtyIdnOrAcc - An optional `PrtyIdn` object containing the party identifier information.
+# + return - Returns an array of strings and/or null values based on the party identifier content and the `fields` 
+# parameter.
+# The array contains specific substrings of the identifier or null values.
+isolated function getPartyIdentifierOrAccount(swiftmt:PrtyIdn? prtyIdnOrAcc) returns string?[] {
+    if prtyIdnOrAcc is swiftmt:PrtyIdn && prtyIdnOrAcc.content.length() > 1 {
+        if prtyIdnOrAcc.content.substring(0, 1).equalsIgnoreCaseAscii("/") {
+            return [(), ...validateAccountNumber(prtyIdn = prtyIdnOrAcc), (), ()];
+        } 
+        if prtyIdnOrAcc.content.startsWith("CUST") || prtyIdnOrAcc.content.startsWith("DRLC") 
+           || prtyIdnOrAcc.content.startsWith("EMPL") {
+            int count = 0;
+            foreach int i in 0... prtyIdnOrAcc.content.length()-1 {
+                if prtyIdnOrAcc.content.substring(i,i+1).equalsIgnoreCaseAscii("/") {
+                    count += 1;
+                }
+                if count == 3 {
+                    return [prtyIdnOrAcc.content.substring(i + 1), (), (), 
+                    prtyIdnOrAcc.content.substring(0, 4), prtyIdnOrAcc.content.substring(8, i)];
+                }
+            }
+            return [prtyIdnOrAcc.content.substring(8), (), (), prtyIdnOrAcc.content.substring(0, 4), ()];
+        } 
+        return [prtyIdnOrAcc.content.toString().substring(8), (), (), prtyIdnOrAcc.content.toString().substring(0, 4), ()];
+    } 
+    return [(), (), (), (), ()];
+}
+
+# Extracts and returns different parts of a party identifier or account information based on its content.
+# The function checks three optional `PrtyIdn` objects (`prtyIdn1`, `prtyIdn2`, and `prtyIdn3`) and determines if they 
+# contain
+# valid identifiers or account numbers. It returns an array of strings where the first element represents the party 
+# identifier, 
+# and the subsequent elements correspond to the account number validation results.
+#
+# + prtyIdn1 - An optional `PrtyIdn` object that may contain a party identifier or account number.
+# + prtyIdn2 - A second optional `PrtyIdn` object that may contain a party identifier or account number.
+# + prtyIdn3 - A third optional `PrtyIdn` object that may contain a party identifier or account number.
+# + return - Returns an array of strings and/or null values based on the party identifier content.
+isolated function getPartyIdentifierOrAccount2(swiftmt:PrtyIdn? prtyIdn1, swiftmt:PrtyIdn? prtyIdn2 = (), swiftmt:PrtyIdn? prtyIdn3 = ()) returns string?[] {
```

Comment:
In this case also we can return a tuple than array, please check other places too.